armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -12,16 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import typing
+import dataclasses
+import functools
+import itertools
+
 from armi import runLog
-from armi.reactor.components import UnshapedComponent
+from armi.reactor.blocks import Block
+from armi.reactor.components import Component, UnshapedComponent
 from armi.reactor.converters.axialExpansionChanger.expansionData import (
-    getSolidComponents,
+    iterSolidComponents,
 )
 
 
-def _determineLinked(componentA, componentB):
+if typing.TYPE_CHECKING:
+    from armi.reactor.assemblies import Assembly
+
+
+def areAxiallyLinked(componentA: Component, componentB: Component) -> bool:
     """Determine axial component linkage for two components.
 
+    Components are considered linked if the following are found to be true:
+
+    1. Both contain solid materials.
+    2. They have compatible types (e.g., ``Circle`` and ``Circle``).
```

Comment:
```suggestion
    2. They are the same types (e.g., ``Circle`` and ``Circle``).
```
Is just a bit more explicit/concrete imo

Reply:
Yes but that's not what we have implemented

https://github.com/terrapower/armi/blob/15e313850c6f71d47f3fde5ec1d047a9fadb7c43/armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py#L49

so you could have A be a subclass of the type of B and be okay.

I'm okay if we want to make that change to `type(A) is type(B)`

Reply:
After discussion - go with `type(A) is type(B)`

Reply:
https://github.com/terrapower/armi/pull/1920/commits/74fcd300def5109ae2a91f91743424904683ca88